consensus/poet/core/tests/test_consensus/test_poet_fork_resolver.py

```diff
@@ -0,0 +1,656 @@
+
+        # change new_fork_mock_wait_certificate duration to a smaller value
+        new_fork_mock_wait_certificate.duration = 0.0
+
+        # set mock_utils.deserialize_wait_certificate
+        # to return a specific value for each fork_head
+        # with cur_fork_head being deserialized first
+        mock_utils.deserialize_wait_certificate.side_effect = \
+            [mock_wait_certificate,
+             new_fork_mock_wait_certificate]
+
+        # check test
+        with mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.'
+                        'LOGGER') as mock_logger:
+            self.assertTrue(fork_resolver.compare_forks(
+                cur_fork_head=mock_cur_fork_header,
+                new_fork_head=mock_new_fork_header))
+
+            # Could be a hack, but verify that the appropriate log message is
+            # generated - so we at least have some faith that the failure was
+            # because of what we are testing and not something else.  I know
+            # that this is fragile if the log message is changed, so would
+            # accept any suggestions on a better way to verify that the
+            # function fails for the reason we expect.
+
+            (message, *_), _ = mock_logger.info.call_args
+            self.assertTrue('New fork wait duration (%f) '
+                            'less than current fork wait duration '
+                            in message)
+
+        # Subtest 3: when new & current fork heads have
+        # the same wait duration
+
+        # change new_fork_mock_wait_certificate duration to a smaller value
+        new_fork_mock_wait_certificate.duration = 1.0
+
+        # set mock_utils.deserialize_wait_certificate
+        # to return a specific value for each fork_head
+        # with cur_fork_head being deserialized first
+        mock_utils.deserialize_wait_certificate.side_effect = \
+            [mock_wait_certificate,
+             new_fork_mock_wait_certificate]
+
+        # check test
+        with mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.'
+                        'LOGGER') as mock_logger:
+            self.assertTrue(fork_resolver.compare_forks(
+                cur_fork_head=mock_cur_fork_header,
+                new_fork_head=mock_new_fork_header))
+
+            # Could be a hack, but verify that the appropriate log message is
+            # generated - so we at least have some faith that the failure was
+            # because of what we are testing and not something else.  I know
+            # that this is fragile if the log message is changed, so would
+            # accept any suggestions on a better way to verify that the
+            # function fails for the reason we expect.
+
+            (message, *_), _ = mock_logger.info.call_args
+            self.assertTrue('New fork header signature (%s) '
+                            'greater than current fork header signature (%s)'
+                            in message)
```

Comment:
You also need to set the `cur_fork_head.header_signature` so that it is greater than `new_fork_head.header_signature` and verify that `cur_fork_head` is chosen.